File: dsa/big-o-complexity-analysis/1.go
package main

func hasDuplicate(nums []int) bool {
	seen := make(map[int]bool)
	for _, n := range nums {
		if seen[n] {
			return true
		}
		seen[n] = true
	}
	return false
}

/*
   R/. Function includes a single loop, so time complexity would be O(n), space complexity, I'm not too sure
*/

File: dsa/big-o-complexity-analysis/1.ts
/*
    Question 1 of 6: what's the time and space complexity of this, and why? Give me both, with reasoning for each.
*/
function hasDuplicate(nums: number[]): boolean {
  const seen = new Set<number>();
  for (const n of nums) {
    if (seen.has(n)) {
      return true;
    }
    seen.add(n);
  }
  return false;
}

/*
    R/. Function includes a single loop, so time complexity would be O(n), space complexity, I'm not too sure
*/
